core/go/internal/sequencer/transport/transport_writer.go

```diff
@@ -0,0 +1,674 @@
+	SendEndorsementResponse(ctx context.Context, transactionId, idempotencyKey, contractAddress string, attResult *prototk.AttestationResult, endorsementResult *components.EndorsementResult, revertReason, endorsementName, party, node string) error
+	SendAssembleRequest(ctx context.Context, assemblingNode string, txID uuid.UUID, idempotencyId uuid.UUID, preAssembly *components.TransactionPreAssembly, stateLocksJSON []byte, blockHeight int64) error
+	SendAssembleResponse(ctx context.Context, txID uuid.UUID, assembleRequestId uuid.UUID, postAssembly *components.TransactionPostAssembly, preAssembly *components.TransactionPreAssembly, recipient string) error
+	SendHandoverRequest(ctx context.Context, activeCoordinator string, contractAddress *pldtypes.EthAddress) error
+	SendNonceAssigned(ctx context.Context, txID uuid.UUID, originatorNode string, contractAddress *pldtypes.EthAddress, nonce uint64) error
+	SendTransactionSubmitted(ctx context.Context, txID uuid.UUID, originatorNode string, contractAddress *pldtypes.EthAddress, txHash *pldtypes.Bytes32) error
+	SendTransactionConfirmed(ctx context.Context, txID uuid.UUID, originatorNode string, contractAddress *pldtypes.EthAddress, nonce *pldtypes.HexUint64, revertReason pldtypes.HexBytes) error
+	SendHeartbeat(ctx context.Context, targetNode string, contractAddress *pldtypes.EthAddress, coordinatorSnapshot *common.CoordinatorSnapshot) error
+	SendPreDispatchRequest(ctx context.Context, originatorNode string, idempotencyKey uuid.UUID, transactionSpecification *prototk.TransactionSpecification, hash *pldtypes.Bytes32) error
+	SendPreDispatchResponse(ctx context.Context, transactionOriginator string, idempotencyKey uuid.UUID, transactionSpecification *prototk.TransactionSpecification) error
+	SendDispatched(ctx context.Context, transactionOriginator string, idempotencyKey uuid.UUID, transactionSpecification *prototk.TransactionSpecification) error
+}
+
+func NewTransportWriter(contractAddress *pldtypes.EthAddress, nodeID string, transportManager components.TransportManager, loopbackHandler func(ctx context.Context, message *components.ReceivedMessage)) TransportWriter {
+	loopbackTransport := NewLoopbackTransportWriter(loopbackHandler)
+	return &transportWriter{
+		nodeID:            nodeID,
+		transportManager:  transportManager,
+		loopbackTransport: loopbackTransport,
+		contractAddress:   contractAddress,
+	}
+}
+
+type transportWriter struct {
+	nodeID            string
+	transportManager  components.TransportManager
+	loopbackTransport LoopbackTransportManager
+	contractAddress   *pldtypes.EthAddress
+}
+
+func (tw *transportWriter) Start(ctx context.Context) error {
+	// We use a separate goroutine to send loopback messages to free up the event loops.
+	go tw.loopbackSend(ctx)
+	return nil
+}
+
+func (tw *transportWriter) SendDelegationRequest(
+	ctx context.Context,
+	coordinatorLocator string,
+	transactions []*components.PrivateTransaction,
+	blockHeight uint64,
+) error {
```

Comment:
Overall feels a bit clunky to have struct defs for these (noting comment on `protobuf` being a potential good way to generate those and have strong commented and typed efficient transfer standards for our protocol), but then not to use them as the input/output format for the code, but rather have flat long lists of input parameters to functions.